cl_nqdemo.c

```diff
@@ -291,10 +325,37 @@ static void NQD_ParseUpdatecolors (void)
 	// fill in userinfo values
 	top = min(colors & 15, 13);
 	bottom = min((colors >> 4) & 15, 13);
+
+	if (cl_demoteamplay.integer)
+		cl.players[i].spectator = (top == 0 && bottom == 0 && cl.players[i].frags <= -99);
```

Comment:
Would be better to have this as an `is_spectator` function... You use the same code earlier as well